Dart/src/com/jetbrains/lang/dart/analyzer/DartAnalysisServerService.java

```diff
@@ -225,9 +230,43 @@ public void run(@NotNull ProgressIndicator indicator) {
     @Override
     public void beforeDocumentChange(DocumentEvent e) {
       myServerData.onDocumentChanged(e);
+      synchronized (myLock) {
+        final Document document = e.getDocument();
+        myUpdatedDocuments.add(document);
+        scheduleUpdateFilesContent();
+      }
     }
   };
 
+  private void scheduleUpdateFilesContent() {
```

Comment:
We seem to already have a mechanism to update file highlighting if it is visible in the editor, after any change, be it external or just typing. Annotators will be started by highlighting daemon after some delay. I'll try to plug into DartServerErrorsAnnotator.collectInformation() instead. Does it sound good?